config/routes.rb

```diff
@@ -2,6 +2,8 @@
   root controller: :home, action: :index
 
   devise_for :users
-
+  resources :subscription, only: [:new]
   resources :articles, only: [:new, :index, :create, :show]
+  root controller: :home, action: :index
+  post '/charges', controller: :subscription, action: :create
```

Comment:
delete this line and update `resources :subscription, only: [:new, :create]` and in your view use the post route for subscriptions